File: stockbot/api/routes/broker_routes.py
from fastapi import APIRouter, Request
from api.controllers import broker_controller
from api.models.broker_models import PortfolioRequest

router = APIRouter()


@router.post("/portfolio")
async def get_portfolio_data(req: PortfolioRequest):
    return await broker_controller.get_portfolio_data(req.broker, req.credentials)

File: backend/routes/brokerRoutes.js
import express from 'express';
import { getActiveBrokerPortfolio } from '../controllers/brokerController.js';
import { protectRoute } from '../middleware/protectRoute.js';

const router = express.Router();

router.get('/portfolio', protectRoute, getActiveBrokerPortfolio);

export default router;
